New Resource: `azurerm_automation_software_update_configuration` (#17902)
hashicorp/terraform-provider-azurerm: website/docs/r/automation_software_update_configuration.html.markdown

---
subcategory: "Automation"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_automation_software_update_configuration"
description: |-
  Manages a Automation.
---

# azurerm_automation_software_update_configuration

Manages a Automation.

## Example Usage

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "example-rg"
  location = "East US"
}

resource "azurerm_automation_account" "test" {
  name                = "example"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  sku_name            = "Basic"

> }
> resource "azurerm_automation_software_update_configuration" "example" {

Reviewer: ```suggestion
}

resource "azurerm_automation_software_update_configuration" "example" {
```

  name                    = "example"
  resource_group_name     = azurerm_resource_group.test.name
  automation_account_name = azurerm_automation_account.test.name

>   operating_system        = "Linux"
>   linux {

Reviewer: ```suggestion
  operating_system        = "Linux"

  linux {
```

    classification    = "Security"
    excluded_packages = ["apt"]
    included_packages = ["vim"]
    reboot_setting    = "IfRequired"

>   }
>   duration = "PT2H2M2S"

Reviewer: ```suggestion
  }

  duration = "PT2H2M2S"
```

}
```

## Arguments Reference

The following arguments are supported:

> * `name` - (Required) The name which should be used for this Automation. Changing this forces a new Automation to be created.

Reviewer: for this automation?

* `automation_account_name` - (Required) The name of the automation account in which the Schedule is created. Changing this forces a new resource to be created.

> * `resource_group_name` - (Required) The name of the Resource Group where the Automation should exist. Changing this forces a new Automation to be created.

Reviewer: these can become `automation_account_id`?

Author: yes, they can. but all other existing resources under Automation are using these two properties to specify the automation account. so I keep this resource the same as others.

Reviewer: They should change to use ID, so can we place use ID here and maybe add comments to the other ones that need to be updated

* `targets` - (Optional) One or more `targets` blocks as defined below.

* `post_task` - (Optional) One or more `post_task` blocks as defined below.

* `pre_task` - (Optional) One or more `pre_task` blocks as defined below.

* `schedule` - (Optional) One or more `schedule` blocks as defined below.

---

A `linux` block supports the following:

* `classification` - (Optional) Specifies the update classifications included in the Software Update Configuration. Possible values are `Unclassified`, `Critical`, `Security` and `Other`.

* `excluded_packages` - (Optional) Specifies a list of packages to excluded from the Software Update Configuration.

* `included_packages` - (Optional) Specifies a list of packages to included from the Software Update Configuration.

* `reboot_setting` - (Optional) Specifies the reboot settings after software update.

---

A `windows` block supports the following:

* `classification` - (Optional) Specifies the update classification. Possible values are `Unclassified`, `Critical`, `Security`, `UpdateRollup`, `FeaturePack`, `ServicePack`, `Definition`, `Tools` and `Updates`.

> * `excluded_kbs` - (Optional) Specifies a list of KB numbers excluded.
> 
> * `included_kbs` - (Optional) Specifies a list of KB numbers included.

Reviewer: kbs is not clear? could we make this
```suggestion
* `excluded_knowledge_base_numbers` - (Optional) Specifies a list of knowledge base numbers excluded.

* `included_knowledge_base_numbers` - (Optional) Specifies a list of knowledge base numbers included.
```

we should avoid acronyms when possible

> * `reboot_setting` - (Optional) Specifies the reboot settings after software update.

Reviewer: what is a reboot setting"?

Author: I have found some other docs about reboot settings (like [spec/compute.json](https://github.com/Azure/azure-rest-api-specs/blob/79bf6bf40f60b4c0c715b389f1e0aca47ce32572/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2020-12-01/compute.json#L8076) and I updated this document by this reference spec.

---

A `targets` block supports the following:

* `azure_queries` - (Optional) One or more `azure_queries` blocks as defined above.

* `non_azure_queries` - (Optional) One or more `non_azure_queries` blocks as defined above.

---

A `azure_queries` block supports the following:

* `locations` - (Optional) Specifies a list of locations to scope the query to.

* `scope` - (Optional) Specifies a list of Subscription or Resource Group ARM Ids to query.

> * `tag_filter` - (Optional) Specifies the tag filter VMs. Possible values are `Any` and `All`.

Reviewer: tag filter VMs?

Author: sorry for my simple using the swagger description here. I have updated this line, please have a look if it's proper now.

---
subcategory: "Automation"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_automation_software_update_configuration"
description: |-

>   Manages a Automation Software Update Configuration.

Reviewer: grammer
```suggestion
  Manages an Automation Software Update Configuration.
```

---

# azurerm_automation_software_update_configuration

> Manages a Automation Software Update Configuraion.

Reviewer: ```suggestion
Manages an Automation Software Update Configuraion.
```

* `resource_group_name` - (Required) The name of the Resource Group where the Automation should exist. Changing this forces a new Automation Software Update Configuration to be created.

* `operating_system` - (Required) The Operating system of target machines. Possible values are `Windows` and `Linux`.

---

* `duration` - (Optional) Maximum time allowed for the software update configuration run. using format `PT[n]H[n]M[n]S` as per ISO8601.

* `linux` - (Optional) One or more `linux` blocks as defined below.

* `windows` - (Optional) One or more `windows` blocks as defined below.

* `virtual_machines` - (Optional) Specifies a list of azure resource Ids of azure virtual machines.

* `non_azure_computer_names` - (Optional) Specifies a list of names of non-azure machines for the software update configuration.

* `targets` - (Optional) One or more `targets` blocks as defined below.

* `post_task` - (Optional) One or more `post_task` blocks as defined below.

* `pre_task` - (Optional) One or more `pre_task` blocks as defined below.

* `schedule` - (Optional) One or more `schedule` blocks as defined below.

---

A `linux` block supports the following:

> * `classification` - (Optional) Specifies the update classifications included in the Software Update Configuration. Possible values are `Unclassified`, `Critical`, `Security` and `Other`.

Reviewer: to make this explicit
```suggestion
* `classifications_included` - (Optional) Specifies the update classifications included in the Software Update Configuration. Possible values are `Unclassified`, `Critical`, `Security` and `Other`.
```

* `excluded_packages` - (Optional) Specifies a list of packages to excluded from the Software Update Configuration.

* `included_packages` - (Optional) Specifies a list of packages to included from the Software Update Configuration.

> * `reboot_setting` - (Optional) Specifies the reboot settings after software update, possible values are `IfRequired`, `Never` and `Always`

Reviewer: i think setting is redundant and slightly confusing so
```suggestion
* `reboot` - (Optional) Specifies the reboot settings after software update, possible values are `IfRequired`, `Never` and `Always`
```

---

A `windows` block supports the following:

* `classification` - (Optional) Specifies the update classification. Possible values are `Unclassified`, `Critical`, `Security`, `UpdateRollup`, `FeaturePack`, `ServicePack`, `Definition`, `Tools` and `Updates`.

* `excluded_knowledge_base_numbers` - (Optional) Specifies a list of knowledge base numbers excluded.

* `included_knowledge_base_numbers` - (Optional) Specifies a list of knowledge base numbers included.

> * `reboot_setting` - (Optional) Specifies the reboot settings after software update, possible values are `IfRequired`, `Never` and `Always`

Reviewer: and here
```suggestion
* `reboot` - (Optional) Specifies the reboot settings after software update, possible values are `IfRequired`, `Never` and `Always`
```
